### DISCUSSION TITLE
ADO.NET - OrleansQuery table created in default database schema (commonly - dbo)

### DISCUSSION THREAD

**Discussion by devbased** [dotnet/orleans#8044] [Q&A] [answered]
And there no way to override it.
Am i missing something? What's the reasoning behind this?

Thank you in advance

**Comment by ReubenBond** [comment, marked as answer]
I don't think you're missing anything. It looks like this something which should be customizable. Any thoughts on the matter, @veikkoeeva or others?
Code: https://github.com/dotnet/orleans/blob/b6beb7fb82a1267f8469c66a28ab756b99ae302c/src/AdoNet/Shared/Storage/DbStoredQueries.cs#L49

**Reply by ElanHasson** [reply]
This should be fully customizable by modifying the SQL scripts prior to deployment.

The ADO.NET storage provider reads all queries except the one called out above from the OrleansQuery table.

**Comment by veikkoeeva** [comment]
The tables can be created in any schema or database. Though Orleans need to know where to read the queries so it can query that database.

For SQL Server one could set the default schema like `ALTER USER [SomeOrleansService] WITH DEFAULT_SCHEMA = [SomeOrleansDeploymentSchema];`. I think this should work. Since that particular query does not define a schema or database or anything on purpose so it can be set from outside.

This issue comes from the earliest of Orleans that did not allow other than define a connection string. Now it would be possible to set schema via options (edit: it needs to be surfaced), for instance (also submit queries and others via other methods than defining them in the DB, changing dynamically while running a cluster could have value, f.ex.).

**Reply by ReubenBond** [reply]
@veikkoeeva it sounds like you are saying that the change proposed in #8061 is not necessary, is that right?

**Reply by veikkoeeva** [reply]
The change proposed would be good if implemented and the idea is to surface the options to Orleans public surface.

So this piece of advice was how to do it if the change proposed were not present.

*To explain more*

This refactoring adds options and in this case I would say it is even better to have these two options. Arguably setting a default schema via database is more or less "hidden" option. While of course setting schemas, accounts etc. is good security and management posture.

One thing is to add callback lamdas for all kinds of purposes. One example being the flag if the connector library is truly asynchronous or not. For instance, https://github.com/dotnet/orleans/blob/main/src/AdoNet/Shared/Storage/DbConstantsStore.cs#L43 should probably be changed. Probably no one is using that specific version of the connector. But the option seem to be useful (at least for testing and to rule out bugs), e.g. there are occasional bugs or then the official connector from Oracle seem to be synchronous.
